lib/web_ui/lib/src/engine/text_editing/text_editing.dart

```diff
@@ -1295,12 +1339,6 @@ abstract class DefaultTextEditingStrategy implements TextEditingStrategy {
     }
   }
 
-  void handleCompositionUpdate(html.Event event) {
-    final EditingState newEditingState = EditingState.fromDomElement(activeDomElement);
-    editingDeltaState.composingOffset = newEditingState.baseOffset!;
-    editingDeltaState.composingExtent = newEditingState.extentOffset!;
-  }
```

Comment:
I honestly don't know how delta state works with composing range. Your change here seems to break delta state because it won't be able to take composing range into account anymore:
https://github.com/flutter/engine/blob/593140b9ce4207fec890c22e4b382ff92e9b389d/lib/web_ui/lib/src/engine/text_editing/text_editing.dart#L506-L512

I'd love to get @Renzo-Olivares's opinion on this one.

Reply:
It seems to be causing some issues, but i'm still looking into it. Ideally we could just pull the `composingOffset` and `composingExtent` from the `EditingState` after `determineCompositionState()` is called, but I'm getting some weird behavior doing that. I'll keep investigating, and post an update asap.

Reply:
We were previously setting the `deltaStart` and `deltaEnd` to the composing region captured in `compositionupdate`. I previously derived the composing region by grabbing the editing state from the DOM at the time when the `compositionupdate` event was fired and using the selection as the composing region. At the time of firing the `compositionupdate` event the editing states selection contains the range was replaced during composition. Previously this worked fine because we were replacing the replaced range with the composed text. However, this was a bit inaccurate since it is not the composing region after the edit was made. When making the composing region accurate through this change, a failure occurs in the _replace method because the range being replaced did not exist yet. For example when composing:

```
Type: ’n’
oldText = ‘’
deltaText = ’n’
deltaStart = ‘0’ (from composingOffset)
deltaEnd = ‘1’ (from composingExtent)
```

`_replace` tries to replace `(0,1)` with ’n’ in oldText which is an empty string. This fails and a delta is not produced. The reason for this being, by deriving the composing region from `compositionupdate` using `CompositionAwareMixin` we are actually deriving the composing region after that composing step has finished, i.e. after ’n’ has already been added to the `EditingState` the composing region is (0,1).

To fix this the `deltaStart` and `deltaEnd` should be the range replaced by the composing region, and not the composing region after the `compositionupdate` event has updated the editing state. We can do this by setting the `deltaStart` to the `composingOffset` captured by the `CompositionAwareMixin` and leaving the `deltaEnd` as the one captured by `beforeInput`. This will be the position of the caret before the `compositionupdate` event is fired. In the case of our example above it would be (0). So coming back to our previous example we would have: 

```
Type: ’n’
oldText = ‘’
deltaText = ’n’
deltaStart = ‘0’ (from composingOffset)
deltaEnd = ‘0’ (from beforeInput extentOffset)
```

Which would not cause a failure on `_replace` and generate the correct delta.

The fix would look something like https://github.com/Renzo-Olivares/engine/commit/f459888fb8275db14a8bb67a32973a65bc24bf5d . 

I'm unsure if we want to rely on 'beforeInput' or if we want to rely on 'CompositionAwareMixin' to save the selection range from the `active DOM` editing state in `compositionupdate` and set the composing region and `deltaStart` and `deltaEnd` through something like `determineCompositionStateForDelta()`.